File: python/simperium/changes.py

class ChangeProcessor(object):
    def process(self, change):
        if change['o'] == 'M':
            if 'sv' in change:
                for key in change['v']:
                    handler = getattr(self, 'on_change_%s' % key, None)
                    if handler:
                        handler(change['d'][key])

File: ruby/lib/simperium/changes.rb
module Simperium
    class ChangeProcessor
        def process(change)
            if change['o'] == 'M'
                if change.include?('sv')
                    change['v'].each do |key|
                        handler = self.send('on_change_#{key}')
                        if handler
                            handler(change['d'][key])
                        end
                    end
                end
            end
        end
    end
end